clippy.sh

```diff
@@ -0,0 +1,113 @@
+    -A clippy::extra_unused_lifetimes \
+    -A clippy::extra_unused_type_parameters \
+    -A clippy::field_reassign_with_default \
+    -A clippy::filter_next \
+    -A clippy::from_over_into \
+    -A clippy::get_first \
+    -A clippy::identity_op \
+    -A clippy::if_same_then_else \
+    -A clippy::inherent_to_string \
+    -A clippy::into_iter_on_ref \
+    -A clippy::iter_nth \
+    -A clippy::iter_nth_zero \
+    -A clippy::large_enum_variant \
+    -A clippy::len_without_is_empty \
+    -A clippy::len_zero \
+    -A clippy::let_and_return \
+    -A clippy::let_unit_value \
+    -A clippy::manual_find \
+    -A clippy::manual_flatten \
+    -A clippy::manual_map \
+    -A clippy::manual_non_exhaustive \
+    -A clippy::manual_range_contains \
+    -A clippy::manual_split_once \
+    -A clippy::manual_strip \
+    -A clippy::manual_unwrap_or \
+    -A clippy::map_clone \
+    -A clippy::match_like_matches_macro \
+    -A clippy::match_ref_pats \
+    -A clippy::missing_safety_doc \
+    -A clippy::module_inception \
+    -A clippy::needless_bool \
+    -A clippy::needless_borrow \
+    -A clippy::needless_borrowed_reference \
+    -A clippy::needless_lifetimes \
+    -A clippy::needless_question_mark \
+    -A clippy::needless_range_loop \
+    -A clippy::needless_return \
+    -A clippy::neg_multiply \
+    -A clippy::never_loop \
+    -A clippy::new_ret_no_self \
+    -A clippy::new_without_default \
+    -A clippy::nonminimal_bool \
+    -A clippy::not_unsafe_ptr_arg_deref \
+    -A clippy::op_ref \
+    -A clippy::option_as_ref_deref \
+    -A clippy::option_map_unit_fn \
+    -A clippy::print_in_format_impl \
+    -A clippy::ptr_arg \
+    -A clippy::question_mark \
+    -A clippy::redundant-async-block \
+    -A clippy::redundant_clone \
+    -A clippy::redundant_closure \
+    -A clippy::redundant_field_names \
+    -A clippy::redundant_pattern \
+    -A clippy::redundant_pattern_matching \
+    -A clippy::redundant_static_lifetimes \
+    -A clippy::result_unit_err \
+    -A clippy::search_is_some \
+    -A clippy::single_char_add_str \
+    -A clippy::single_char_pattern \
+    -A clippy::single_component_path_imports \
+    -A clippy::single_match \
+    -A clippy::suspicious_to_owned \
+    -A clippy::too_many_arguments \
+    -A clippy::type_complexity \
+    -A clippy::unit_arg \
+    -A clippy::unnecessary_cast \
+    -A clippy::unnecessary_lazy_evaluations \
+    -A clippy::unnecessary_mut_passed \
+    -A clippy::unnecessary_sort_by \
+    -A clippy::unnecessary_to_owned \
+    -A clippy::unused_io_amount \
+    -A clippy::unused_unit \
+    -A clippy::unwrap_or_else_default \
+    -A clippy::upper_case_acronyms \
+    -A clippy::useless_conversion \
+    -A clippy::useless_format \
+    -A clippy::vec_box \
+    -A clippy::while_let_on_iterator \
+    -A clippy::wildcard_in_or_patterns \
+    -A clippy::write_with_newline \
+    -A clippy::wrong_self_convention \
+    -A clippy::zero_prefixed_literal \
```

Comment:
Oh boy, that's a list!

Let's have some straight talk about this; I appreciate the time and energy you've invested here, but in looking at this list, all I see is ongoing work.

Software engineering work falls into a couple of categories:

* Implementing features
* Fixing bugs
* Improving infrastructure so that either:
  * Features are easier to implement
  * bugs are harder to create

There's value in all of those.  The first two are the highest value: they are most noticed by the users. The last two are often invisible to users and their value is less tangible.

Executing an engineering project is primarily about making informed choices on how to spend the available time, and that typically means
picking work with a good effort:reward ratio.

Glancing at this list of lints, I see over 100 discrete things that need to be kept track of in some way, wether it is an allow-list that aspires to get longer or a deny-list that aspires to get shorter, it requires at least 1 human to understand each entry on that list in order to be take on maintenance of it. If you estimate 30 minutes per human per item to review/coordinate, that's over 50 hours of work, which is a significant chunk of time.

My opinion is that most of the lints are stylistic choices that don't meaningfully improve the code cleanliness over its current state, and don't help to prevent the kinds of bugs that are most likely to crop up in a real application.

Importantly, none of the clippy lints in the PRs you've opened have found any real bugs or serious performance issues, which suggests that everything of value that it could find has already been found, which in turn suggests that there are diminishing returns on building infrastructure to support clippy.

That makes it hard to justify the time.

Since this is a spare-time project it's even harder to justify: there's no financial pay off that will let either of us say "well, that sucked, but at least I earned something!" at the end of it.

And yes, sometimes you just want to work on something for personal reasons: maybe it's for fun, maybe you're pushing the bounds of your knowledge, and that's OK when it's just your own time that you're spending. When you need to get the result merged into someone else's project it takes up their time both for the review and for the ongoing maintenance of your contribution.

I can't justify spending my spare time on shepherding these clippy changes any further: the value to me and this project is too low and I have too many other things that are higher priority.

So: all of this is a really long way of saying: thank you for contributing, but I don't want you to burn more of your own time on this because I don't think it is worth going further.

Sorry!

Reply:
Thanks for taking the time ti write this, I appreciate it very much.

This was a learning experience which helped me understand more about the codebase, I will take some time to reflect on it, but I already have some ideas on higher value contributions, specifically failing CI faster, so that when something is wrong, checks don't continue running.

Again, thank you for taking this time to reply, I am very happy about it.